### ISSUE TITLE
patch_emscripten.py doesn't return error code when failing

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot-mono-builds#6]
I noticed that in my current builds of mono-6.6.0.166 with emscripten 1.38.47-upstream, the `patch_emscripten.py` command actually has some failed hunks, but the build script continues nevertheless.

```
[root@e381878f0b5a godot-mono-builds]# python3 patch_emscripten.py 
[root@e381878f0b5a godot-mono-builds]# patching file src/support.js
Reversed (or previously applied) patch detected!  Skipping patch.
1 out of 1 hunk ignored -- saving rejects to file src/support.js.rej
patching file emcc.py
patching file tests/test_browser.py
Reversed (or previously applied) patch detected!  Skipping patch.
1 out of 1 hunk ignored -- saving rejects to file tests/test_browser.py.rej
Hunk #1 succeeded at 930 with fuzz 2 (offset 40 lines).
Hunk #2 FAILED at 918.
Hunk #3 FAILED at 987.
Hunk #4 FAILED at 1841.
3 out of 4 hunks FAILED -- saving rejects to file emcc.py.rej
echo $?
0
```

It should return a non-0 value when failing.

There's also some problems with newlines/early return which might explain why it returns 0 even though some of the patch calls may fail. (See how the prompt returns right after the `python3 patch_emscripten.py` call.)

**Comment by akien-mga** [comment]
Seems like you use `exit 0` explicitly:
https://github.com/godotengine/godot-mono-builds/blob/master/patch_emscripten.py#L32

**Comment by neikeq** [comment]
Fixed in 27b5523a2d93d84d3472a8feec0df650d5f5f0b9
